### PR TITLE
Change: Always generate API files at build time

### PR DESCRIPTION
**Pull request by glx22**
Running API script is often forgotten. This PR ensure API files are always up to date.

### CODE REVIEW

File: src/ai/ai_instance.cpp.in [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+
+#include "../script/script_storage.hpp"
+#include "ai_info.hpp"
+#include "ai_instance.hpp"
+
+/* Manually include the Text glue. */
+#include "../script/api/template/template_text.hpp.sq"
+
+/* Convert all AI related classes to Squirrel data.
+ * Note: this line is a marker in squirrel_export.sh. Do not change! */
```
Comment (LordAro) [inline src/ai/ai_instance.cpp.in]: Comment could do with an update

File: src/ai/ai_instance.cpp.in [hunk 2]
```diff
@@ -0,0 +1,121 @@
+/*
```
Comment (LordAro) [inline src/ai/ai_instance.cpp.in]: Perhaps instead of autogenerating the entire file, it could include a "ai_includes.hpp" file that's built by cmake?

**Review by LordAro — DISMISSED** [review]
Mm, 667 lines cmake file. Nice. (I'd like to see someone else approve it before merging)

**Comment by glx22** [comment]
I have some doubt about doxygen stuff. Not sure it's still working as script_window.hpp is now "outside" source.

**Comment by glx22** [comment]
I decided to go the easy way to fix doxygen stuff. Will need to update compile farm for it.

**Review by LordAro — APPROVED** [review]

**Pull request merged**